src/OrleansRuntime/Streams/PubSub/PubSubRendezvousGrain.cs

```diff
@@ -466,20 +407,19 @@ public Task<List<StreamSubscription>> GetAllSubscriptions(StreamId streamId, ISt
             {
                 RemoveProducer(producer);
             }
-        }
-
-        /// <summary>
-        /// Try clear state will only clear the state if there are no producers or consumers.
-        /// </summary>
-        /// <returns></returns>
-        private async Task<bool> TryClearState()
-        {
-            if (State.Producers.Count == 0 && State.Consumers.Count == 0) // + we already know that numProducers == 0 from previous if-clause
+            catch (OrleansMessageRejectionException)
```

Comment:
Let's ensure that `SiloUnavailableException` is also handled here, since it signifies that a silo was determined to be dead by membership while some request was outstanding.